utils/s2n_fork_detection.c

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/* This captures Darwin specialities. This is the only APPLE flavor we care about.
+ * Here we also capture varius required feature test macros.
+ */
+#if defined(__APPLE__)
+    typedef struct _opaque_pthread_once_t  __darwin_pthread_once_t;
+    typedef __darwin_pthread_once_t pthread_once_t;
+    #define _DARWIN_C_SOURCE
+#elif !defined(_GNU_SOURCE)
+    /* Keep in sync with feature probe tests/features/madvise.c */
+    #define _GNU_SOURCE
+#endif
+
+#include <sys/mman.h>
+
+/* Not always defined for Darwin */
+#if !defined(MAP_ANONYMOUS)
+    #define MAP_ANONYMOUS MAP_ANON
+#endif
+
+#include "error/s2n_errno.h"
+#include "utils/s2n_fork_detection.h"
+#include "utils/s2n_safety.h"
+
+#include <pthread.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+
+#if defined(S2N_MINHERIT_SUPPORTED) && defined(S2N_MINHERIT_SUPPORTED)
+#error "Both S2N_MINHERIT_SUPPORTED and S2N_MINHERIT_SUPPORTED are defined. This should not be possible."
+#endif
```

Comment:
I'm confused... This is the same `S2N_MINHERIT_SUPPORTED` variable?

Reply:
That's my bad. Must have gone wild with copy-paste when adding the prefix. 

I went through the other macro's to check that they are okay.